pkg/kubelet/pod_workers.go

```diff
@@ -620,12 +629,18 @@ func (p *podWorkers) UpdatePod(options UpdatePodOptions) {
 	var podUpdates chan podWork
 	var exists bool
 	if podUpdates, exists = p.podUpdates[uid]; !exists {
+		fullname := kubecontainer.GetPodFullName(pod)
+		if p.isConflictedStaticPod(pod) {
```

Comment:
only the same static pod， while it's SYNC work, it still enters this condition, that's not correct. so this condition maybe not enough

Reply:
Can you elaborate on this? Which condition do you think we need to add?

We are trying to ensure that there is only one running static pod with the same full name.

Reply:
this condition is if `p.knownPodFullnames[fullName]` already has the pod full name, it considers as conflict,  but with the same static pod, it still enter the process periodly for it's sync interval, so `p.reservedPodUIDsFullnames[fullname]` may store many unuseful uids.

Reply:
> but with the same static pod, it still enter the process periodly for it's sync interval,

I think the same(having the same uid) static pod cannot enter here twice. After a static pod enters here, podUpdates[uid] exists for the same pod, so we can't enter again for the same static pod.

Please let me know if I misunderstand.

Reply:
oh sorry, it's my mistake